assets/app/actions/siteActions.js

```diff
@@ -55,6 +55,23 @@ export default {
     }).catch(error => alertActions.httpError(error.message));
   },
 
+  uploadFile(site, path, fileData, message) {
```

Comment:
Moved file parsing to the action creator. We still can't unify both of the functions for now, as the file is going to be run through a FileReader object, which encodes the contents to bas64 for us.